### ISSUE TITLE
Statistical robustness: train/held-out split, significance testing, early stopping

### ISSUE THREAD

**Issue by anibjoshi** [stratalab/strata-core#2309]
## Context

With iterative recipe optimization (10–20 rounds of agent/optimizer-proposed recipes), the eval set is being adaptively queried. Without statistical safeguards, the winning recipe is likely overfit to the eval set.

## Why

50 eval queries can only detect differences ≥ 0.06–0.10 nDCG — barely enough to distinguish meaningful improvements from noise when recipe tuning provides 5–10% total headroom. Voorhees (SIGIR 2009) showed even TREC's 50 topics produce 5–10% swap rates (wrong system declared winner). With 20 rounds × 5 recipes against the same 50 queries, Freedman's rule puts us at p/n = 0.6 — deep in the overfitting danger zone.

### 1. Automatic train/held-out split

```rust
fn split_eval_set(eval_set: &[EvalQuery], held_out_ratio: f64) -> (Vec<EvalQuery>, Vec<EvalQuery>) {
    let split_point = (eval_set.len() as f64 * (1.0 - held_out_ratio)) as usize;
    let mut shuffled = eval_set.to_vec();
    shuffled.shuffle(&mut rng);
    (shuffled[..split_point].to_vec(), shuffled[split_point..].to_vec())
}
```

- Default held_out_ratio = 0.3
- Optimizer and agent see ONLY the optimization set
- Final winner evaluated against held-out set for unbiased estimate
- Report both optimization-set and held-out-set metrics

### 2. Paired bootstrap significance testing

For every recipe comparison, compute p-value via paired bootstrap:

```rust
fn paired_bootstrap_test(
    scores_a: &[f64],  // per-query nDCG for recipe A
    scores_b: &[f64],  // per-query nDCG for recipe B
    n_bootstrap: usize, // 10,000
) -> f64 {
    let observed_diff = mean(scores_a) - mean(scores_b);
    let mut count_ge = 0;
    for _ in 0..n_bootstrap {
        let resampled = bootstrap_resample(scores_a, scores_b);
        if resampled_diff >= observed_diff { count_ge += 1; }
    }
    count_ge as f64 / n_bootstrap as f64
}
```

Include in rankings output:
```json
{
  "name": "tuned",
  "ndcg@10": 0.71,
  "vs_baseline_p": 0.003,
  "confidence_interval_95": [0.67, 0.75]
}
```

### 3. Early stopping for inferior recipes

During experiment execution, if Recipe A's confidence interval doesn't overlap Recipe B's after 30 queries, stop the inferior recipe:

```rust
fn should_stop_early(running_ci_a: (f64, f64), running_ci_b: (f64, f64)) -> bool {
    running_ci_a.0 > running_ci_b.1  // A's lower bound > B's upper bound
}
```

Netflix's adaptive experimentation approach. Saves 30–50% of evaluation compute.

### 4. Overfitting monitoring

Track gap between optimization-set and held-out-set performance:
```json
{
  "optimization_set_ndcg": 0.71,
  "held_out_ndcg": 0.65,
  "overfit_gap": 0.06,
  "warning": "Possible overfitting: held-out score 8.5% below optimization score"
}
```

Flag when gap exceeds 5%.

### Minimum requirements

| Component | Minimum | Recommended |
|---|---|---|
| Total eval queries | 100 | 200+ |
| Optimization set | 70 | 140 |
| Held-out test set | 30 | 60 |
| Max optimization rounds | 10 | 15 with early stopping |
| Significance threshold | p < 0.01 | p < 0.05 with Holm correction |

## References

- Voorhees, "Topic Set Size Redux", SIGIR 2009 (5–10% swap rates at n=50)
- Smucker et al., CIKM 2007 (bootstrap ≈ paired t-test for n ≥ 50)
- Freedman's rule: overfitting ~ p/n
- `docs/design/search/optimization-research.md` §3
